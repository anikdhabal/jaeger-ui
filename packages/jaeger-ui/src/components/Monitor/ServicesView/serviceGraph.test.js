// Copyright (c) 2021 The Jaeger Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import { shallow } from 'enzyme';
import ServiceGraph, { tickFormat } from './serviceGraph';
import { serviceMetrics } from '../../../reducers/metrics.mock';

const props = {
  width: 300,
  error: null,
  name: 'Hello Graph',
  metricsData: null,
  loading: true,
  marginClassName: '',
  crosshairValues: [],
  xDomain: [1, 2],
};

describe('<ServiceGraph>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ServiceGraph {...props} />);
  });

  it('does not explode', () => {
    expect(wrapper.length).toBe(1);
  });

  it('displays loading indicator when loading', () => {
    expect(wrapper.find('LoadingIndicator').exists()).toBe(true);
  });

  it('displays loading indicator when xDomain is empty', () => {
    wrapper.setProps({ ...props, xDomain: [], loading: false });
    expect(wrapper.find('LoadingIndicator').exists()).toBe(true);
  });

  it('displays "No Data" message when metricsData is null', () => {
    wrapper.setProps({ ...props, loading: false });
    expect(wrapper.text()).toContain('No Data');
  });

  it('displays "Couldn’t fetch data" message when error is present', () => {
    wrapper.setProps({ ...props, loading: false, error: new Error('API Error') });
    expect(wrapper.text()).toContain('Couldn’t fetch data');
  });

  it('displays base graph when metricsData is provided', () => {
    wrapper.setProps({
      ...props,
      loading: false,
      metricsData: serviceMetrics.service_call_rate,
    });
    expect(wrapper.find('ComposedChart').exists()).toBe(true);
  });

  it('displays base graph with legends when showLegend is true', () => {
    wrapper.setProps({
      ...props,
      loading: false,
      metricsData: serviceMetrics.service_call_rate,
      showLegend: true,
    });
    expect(wrapper.find('Legend').exists()).toBe(true);
  });

  it('displays base graph with horizontal lines when showHorizontalLines is true', () => {
    wrapper.setProps({
      ...props,
      loading: false,
      metricsData: serviceMetrics.service_call_rate,
      showHorizontalLines: true,
    });
    expect(wrapper.find('CartesianGrid[horizontal=true]').exists()).toBe(true);
  });

  it('displays base graph with custom color', () => {
    wrapper.setProps({
      ...props,
      loading: false,
      metricsData: serviceMetrics.service_call_rate,
      color: 'AAAAAA',
    });
    expect(wrapper.find('Area[stroke="AAAAAA"]').exists()).toBe(true);
    expect(wrapper.find('Line[stroke="AAAAAA"]').exists()).toBe(true);
  });

  it('Tooltip test', () => {
    wrapper.setProps({
      ...props,
      loading: false,
      metricsData: serviceMetrics.service_call_rate,
      crosshairValues: [
        {
          key: 1,
          ...serviceMetrics.service_call_rate.metricPoints[0],
        },
      ],
    });
    expect(wrapper.find('Tooltip').exists()).toBe(true);
  });

  it('Tooltip hover test', () => {
    wrapper.setProps({
      ...props,
      loading: false,
      metricsData: {
        ...serviceMetrics.service_call_rate,
        metricPoints: [{ x: 1631271783806, y: null }, ...serviceMetrics.service_call_rate.metricPoints],
      },
    });
    expect(wrapper.find('Area').length).toBeGreaterThan(0);
    expect(wrapper.find('Line').length).toBeGreaterThan(0);
  });

  it('displays Tooltip with custom formatter', () => {
    wrapper.setProps({
      ...props,
      loading: false,
      metricsData: serviceMetrics.service_call_rate,
      crosshairValues: [
        {
          key: 1,
          ...serviceMetrics.service_call_rate.metricPoints[0],
        },
      ],
    });

    const tooltipComponent = wrapper.find('Tooltip');
    const exampleValue = 42;
    const exampleEntry = {
      payload: {
        x: 1631271783806,
        y: 123,
        label: 0.75,
      },
    };
    const formattedTooltipContent = tooltipComponent.prop('formatter')(exampleValue, 'name', exampleEntry);

    const formattedLabel = new Date(exampleEntry.payload.x).toLocaleTimeString();
    const expectedTooltipContent = [
      `label: ${exampleEntry.payload.label * 100}`,
      `x: ${formattedLabel}`,
      `Y: ${exampleEntry.payload.y}`,
    ];

    expect(formattedTooltipContent).toEqual(expectedTooltipContent);
  });
});

describe('<ServiceGraph> util', () => {
  it('tickFormat singular', () => {
    expect(tickFormat(Date.UTC(2017, 1, 14, 15, 19)).valueOf()).toBe('15:19');
  });

  it('tickFormat plural', () => {
    expect(tickFormat(Date.UTC(2017, 1, 14, 4, 4)).valueOf()).toBe('04:04');
  });
});
