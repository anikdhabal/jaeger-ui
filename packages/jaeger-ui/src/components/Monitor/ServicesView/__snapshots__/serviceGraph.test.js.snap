// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceGraph> "Couldn’t fetch data" displayed 1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <div
    className="center-placeholder"
    style={
      Object {
        "height": 168,
        "width": 300,
      }
    }
  >
    Couldn’t fetch data
  </div>
</div>
`;

exports[`<ServiceGraph> "No data" displayed 1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <div
    className="center-placeholder"
    style={
      Object {
        "height": 168,
        "width": 300,
      }
    }
  >
    No Data
  </div>
</div>
`;

exports[`<ServiceGraph> Base graph should be displayed  1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <ForwardRef
    height={168}
    width={300}
  >
    <ComposedChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "bottom": 25,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#ccc"
        vertical={false}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="x"
        domain={
          Array [
            1,
            2,
          ]
        }
        height={30}
        hide={false}
        interval={5}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={true}
        scale="auto"
        tickCount={5}
        tickFormatter={[Function]}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Area
        activeDot={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={false}
        fill="#DCA3D2"
        fillOpacity={0.1}
        hide={false}
        isAnimationActive={true}
        key="0"
        legendType="line"
        points={Array []}
        stroke="#DCA3D2"
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        key="1"
        label={false}
        legendType="line"
        points={Array []}
        stroke="#DCA3D2"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={
          Object {
            "color": "#6b6b76",
            "width": 140,
          }
        }
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        formatter={[Function]}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        reverseDirection={
          Object {
            "x": false,
            "y": false,
          }
        }
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "height": 0,
            "width": 0,
            "x": 0,
            "y": 0,
          }
        }
        wrapperStyle={Object {}}
      />
    </ComposedChart>
  </ForwardRef>
</div>
`;

exports[`<ServiceGraph> Base graph with custom color should be displayed 1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <ForwardRef
    height={168}
    width={300}
  >
    <ComposedChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "bottom": 25,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#ccc"
        vertical={false}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="x"
        domain={
          Array [
            1,
            2,
          ]
        }
        height={30}
        hide={false}
        interval={5}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={true}
        scale="auto"
        tickCount={5}
        tickFormatter={[Function]}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Area
        activeDot={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={false}
        fill="AAAAAA"
        fillOpacity={0.1}
        hide={false}
        isAnimationActive={true}
        key="0"
        legendType="line"
        points={Array []}
        stroke="AAAAAA"
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        key="1"
        label={false}
        legendType="line"
        points={Array []}
        stroke="AAAAAA"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={
          Object {
            "color": "#6b6b76",
            "width": 140,
          }
        }
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        formatter={[Function]}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        reverseDirection={
          Object {
            "x": false,
            "y": false,
          }
        }
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "height": 0,
            "width": 0,
            "x": 0,
            "y": 0,
          }
        }
        wrapperStyle={Object {}}
      />
    </ComposedChart>
  </ForwardRef>
</div>
`;

exports[`<ServiceGraph> Base graph with horizontal lines should be displayed 1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <ForwardRef
    height={168}
    width={300}
  >
    <ComposedChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "bottom": 25,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#ccc"
        vertical={false}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="x"
        domain={
          Array [
            1,
            2,
          ]
        }
        height={30}
        hide={false}
        interval={5}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={true}
        scale="auto"
        tickCount={5}
        tickFormatter={[Function]}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Area
        activeDot={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={false}
        fill="#DCA3D2"
        fillOpacity={0.1}
        hide={false}
        isAnimationActive={true}
        key="0"
        legendType="line"
        points={Array []}
        stroke="#DCA3D2"
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        key="1"
        label={false}
        legendType="line"
        points={Array []}
        stroke="#DCA3D2"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={
          Object {
            "color": "#6b6b76",
            "width": 140,
          }
        }
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        formatter={[Function]}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        reverseDirection={
          Object {
            "x": false,
            "y": false,
          }
        }
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "height": 0,
            "width": 0,
            "x": 0,
            "y": 0,
          }
        }
        wrapperStyle={Object {}}
      />
    </ComposedChart>
  </ForwardRef>
</div>
`;

exports[`<ServiceGraph> Base graph with legends should be displayed 1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <ForwardRef
    height={168}
    width={300}
  >
    <ComposedChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "bottom": 25,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#ccc"
        vertical={false}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="x"
        domain={
          Array [
            1,
            2,
          ]
        }
        height={30}
        hide={false}
        interval={5}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={true}
        scale="auto"
        tickCount={5}
        tickFormatter={[Function]}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Area
        activeDot={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={false}
        fill="#DCA3D2"
        fillOpacity={0.1}
        hide={false}
        isAnimationActive={true}
        key="0"
        legendType="line"
        points={Array []}
        stroke="#DCA3D2"
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        key="1"
        label={false}
        legendType="line"
        points={Array []}
        stroke="#DCA3D2"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={
          Object {
            "color": "#6b6b76",
            "width": 140,
          }
        }
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        formatter={[Function]}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        reverseDirection={
          Object {
            "x": false,
            "y": false,
          }
        }
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "height": 0,
            "width": 0,
            "x": 0,
            "y": 0,
          }
        }
        wrapperStyle={Object {}}
      />
      <Legend
        align="center"
        formatter={[Function]}
        iconSize={14}
        layout="horizontal"
        margin={
          Object {
            "left": "10px",
          }
        }
        verticalAlign="top"
      />
    </ComposedChart>
  </ForwardRef>
</div>
`;

exports[`<ServiceGraph> Loading indicator is displayed 1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <div
    className="center-placeholder"
    style={
      Object {
        "height": 168,
        "width": 300,
      }
    }
  >
    <LoadingIndicator
      centered={true}
      small={false}
    />
  </div>
</div>
`;

exports[`<ServiceGraph> Loading indicator is displayed when xDomain is empty 1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <div
    className="center-placeholder"
    style={
      Object {
        "height": 168,
        "width": 300,
      }
    }
  >
    <LoadingIndicator
      centered={true}
      small={false}
    />
  </div>
</div>
`;

exports[`<ServiceGraph> Tooltip hover test 1`] = `null`;

exports[`<ServiceGraph> Tooltip hover test 2`] = `null`;

exports[`<ServiceGraph> Tooltip test 1`] = `
<div
  className="graph-container "
  style={
    Object {
      "height": 242,
    }
  }
>
  <h3
    className="graph-header"
  >
    Hello Graph
  </h3>
  <ForwardRef
    height={168}
    width={300}
  >
    <ComposedChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
          Array [
            Object {
              "x": 1631271823806,
              "y": 0.05,
            },
            Object {
              "x": 1631271883806,
              "y": 0.05,
            },
          ],
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "bottom": 25,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#ccc"
        vertical={false}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="x"
        domain={
          Array [
            1,
            2,
          ]
        }
        height={30}
        hide={false}
        interval={5}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={true}
        scale="auto"
        tickCount={5}
        tickFormatter={[Function]}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Area
        activeDot={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={false}
        fill="#DCA3D2"
        fillOpacity={0.1}
        hide={false}
        isAnimationActive={true}
        key="0"
        legendType="line"
        points={Array []}
        stroke="#DCA3D2"
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={true}
        dataKey="y"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        key="1"
        label={false}
        legendType="line"
        points={Array []}
        stroke="#DCA3D2"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={
          Object {
            "color": "#6b6b76",
            "width": 140,
          }
        }
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        formatter={[Function]}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        reverseDirection={
          Object {
            "x": false,
            "y": false,
          }
        }
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "height": 0,
            "width": 0,
            "x": 0,
            "y": 0,
          }
        }
        wrapperStyle={Object {}}
      />
    </ComposedChart>
  </ForwardRef>
</div>
`;
